exp/api/v1beta1/azuremachinepool_default_test.go

```diff
@@ -48,75 +53,138 @@ func TestAzureMachinePool_SetDefaultSSHPublicKey(t *testing.T) {
 }
 
 func TestAzureMachinePool_SetIdentityDefaults(t *testing.T) {
-	g := NewWithT(t)
-
-	type test struct {
-		machinePool *AzureMachinePool
-	}
-
 	fakeSubscriptionID := uuid.New().String()
 	fakeClusterName := "testcluster"
 	fakeRoleDefinitionID := "testroledefinitionid"
 	fakeScope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", fakeSubscriptionID, fakeClusterName)
 	existingRoleAssignmentName := "42862306-e485-4319-9bf0-35dbc6f6fe9c"
-	roleAssignmentExistTest := test{machinePool: &AzureMachinePool{Spec: AzureMachinePoolSpec{
-		Identity: infrav1.VMIdentitySystemAssigned,
-		SystemAssignedIdentityRole: &infrav1.SystemAssignedIdentityRole{
-			Name: existingRoleAssignmentName,
+	machinePool := &expv1.MachinePool{
```

Comment:
I would vote to move these new objects inside the `t.Run` if they aren't referenced in the test cases just to make sure these don't get mutated in one test case such that it causes issues in another test case.

I realize though that here it's fairly obvious that the AzureMachinePool defaulting webhook won't be mutating these other kinds of objects.

Reply:
done